addons/account/models/account_journal.py

```diff
@@ -57,31 +57,32 @@ def _default_alias_domain(self):
     account_control_ids = fields.Many2many('account.account', 'journal_account_control_rel', 'journal_id', 'account_id', string='Accounts Allowed',
         check_company=True,
         domain="[('deprecated', '=', False), ('company_id', '=', company_id)]")
-    default_credit_account_id = fields.Many2one('account.account', string='Default Credit Account', copy=False,
-        domain=[('deprecated', '=', False)], help="It acts as a default account for credit amount",
-        check_company=True,
-        ondelete='restrict')
-    default_debit_account_id = fields.Many2one('account.account', string='Default Debit Account', copy=False,
-        check_company=True,
-        domain="[('deprecated', '=', False), ('company_id', '=', company_id)]", help="It acts as a default account for debit amount", ondelete='restrict')
+    default_account_type = fields.Many2one('account.account.type', compute="_compute_default_account_type")
+    default_account_id = fields.Many2one('account.account', string='Default Account', copy=False,
+                                         domain="[('deprecated', '=', False), ('company_id', '=', company_id),"
+                                                " ('user_type_id', '=', default_account_type)]",
+                                         check_company=True, help="It acts as a default account", ondelete='restrict')
```

Comment:
actually: default value for default_account_type being True (for new records, records without type, or with type == general)  could save the day...

Reply:
agree for the [('user_type_id.type', 'not in', ('receivable', 'payable'))]. 

the attrs instead of the function field is an implementation detail, not sure which would be cleaner :thinking:

Reply:
> What about the exchange difference journal? I don't see accounts anymore and odoo is not able to retrieve the debit/credit accounts: `Exchange difference Loss` & `Exchange difference Gain`.
>> @smetl I must confess I'm not sure I understand the purpose of a misc journal for exchange differences. I don't see the use case... can you tell me an example ?

Reply:
Both accounts are now on the company so it doesn't matter

Reply:
yes it does: if you have several misc journals (OD, OD salaires, wathever,...), how do you choose in which journal the exchange difference entry will be created?

Reply:
@qdp-odoo I still wonder the use case a user might face that justifies a exchange difference gain or exchange difference loss. Isn't it only when there is a transaction (bank & cash) ? The difference occurs when you obtain less/more than the expected amount. So, it requires a use case with a money transfer, no ?

Reply:
indeed.

but that doesn't mean that exchange difference is made in the cash/bank transaction or journal, nor at that time: you make an invoice on D1 100$, record a payment on D2 100$, reconcile the payment and the invoice in D3. That's only on D3 that the currency exchange difference must be recorded (before that, you can't say if there's any to book), so you must create a new entry and can't touch anymore original entries from D1 or D2. And you don't want to steal a number in your bank journal, do you?